ccdproc/ccddata.py

```diff
@@ -466,46 +532,61 @@ def fits_ccddata_reader(filename, hdu=0, unit=None, **kwd):
         if key in kwd:
             prefix = 'Unsupported keyword: {0}.'.format(key)
             raise TypeError(' '.join([prefix, msg]))
-    hdus = fits.open(filename, **kwd)
-    hdr = hdus[hdu].header
-
-    # search for the first instance with data if the primary header is empty
-    if hdu == 0 and hdus[hdu].data is None:
-        for i in range(len(hdus)):
-            if hdus.fileinfo(i)['datSpan'] > 0:
-                hdu = i
-                hdr = hdr + hdus[hdu].header
-                log.info("First HDU with data is exention {0}.".format(hdu))
-                break
-
-    try:
-        fits_unit_string = hdr['bunit']
-        # patch to handle FITS files using ADU for the unit instead of the
-        # standard version of 'adu'
-        if fits_unit_string.strip().lower() == 'adu':
-            fits_unit_string = fits_unit_string.lower()
-    except KeyError:
-        fits_unit_string = None
-
-    if unit is not None and fits_unit_string:
-        log.info("Using the unit {0} passed to the FITS reader instead of "
-                 "the unit {1} in the FITS file.".format(unit,
-                                                         fits_unit_string))
-
-    use_unit = unit or fits_unit_string
-    # Try constructing a WCS object. This may generate a warning, but never
-    # an error.
-    wcs = WCS(hdr)
-    # Test for success by checking to see if the wcs ctype has a non-empty
-    # value.
-    wcs = wcs if wcs.wcs.ctype[0] else None
-    ccd_data = CCDData(hdus[hdu].data, meta=hdr, unit=use_unit,
-                       wcs=wcs)
-    hdus.close()
+    with fits.open(filename, **kwd) as hdus:
```

Comment:
I hope you don't mind that I switched this one to the context manager. I had some problems with Errors and that file handles on Windows are still open because it did not reach the ``hdus.close()``.